### PR TITLE
Add test_socket

### PR DESCRIPTION
**Pull request by coolreader18**

### CODE REVIEW

File: vm/src/builtins/pystr.rs [hunk 1]
```diff
@@ -1168,23 +1171,6 @@ impl IntoPyObject for &String {
...
-
-impl TryFromObject for std::ffi::OsString {
-    fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
-        use std::str::FromStr;
-
-        let s = PyStrRef::try_from_object(vm, obj)?;
-        Ok(std::ffi::OsString::from_str(s.as_str()).unwrap())
-    }
-}
-
```
Comment (fanninpm) [inline vm/src/builtins/pystr.rs:1187]: Were you able to compile this on your machine?

Comment (coolreader18) [reply vm/src/builtins/pystr.rs:1187]: Oh, I guess I didn't check with ssl enabled

File: vm/src/stdlib/socket.rs [hunk 2]
```diff
@@ -47,13 +47,33 @@ macro_rules! errcode {
...
-        SOCK_RDM, SOCK_STREAM, SOL_SOCKET, SO_BROADCAST, SO_ERROR, SO_LINGER, SO_OOBINLINE,
-        SO_REUSEADDR, SO_TYPE, *,
+        SOCK_RDM, SOCK_SEQPACKET, SOCK_STREAM, SOL_SOCKET, SO_BROADCAST, SO_ERROR, SO_LINGER,
+        SO_OOBINLINE, SO_REUSEADDR, SO_TYPE, *,
     };
 }
+#[cfg(windows)]
+use winapi::shared::netioapi;
+
+fn get_raw_sock(obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<RawSocket> {
```
Comment (fanninpm) [inline vm/src/stdlib/socket.rs:62]: Did you mean `get_raw_socket`?

**Review by youknowone — APPROVED** [review]

**Comment by youknowone** [comment]
@fanninpm @coolreader18 is there unfinished further work here?

**Comment by DimitrisJim** [comment]
Referencing #1176 for others to be aware of this PR.

**Comment by coolreader18** [comment]
Nope, this is done now!

File: vm/src/builtins/pystr.rs (hunk 1 above)
Comment (DimitrisJim) [reply vm/src/builtins/pystr.rs:1187]: Ah, #2707 used this to convert into a `String` and that got in after the tests ran for your PR. I was wondering why the tests passed fine in the PR but not when it was merged.

Comment (fanninpm) [reply vm/src/builtins/pystr.rs:1187]: > I was wondering why the tests passed fine in the PR but not when it was merged.

Bors could’ve probably helped.

**Pull request merged**